### PR TITLE
test for construct functions

### PR DESCRIPTION
**Pull request by tylerflex**
based on my tests, it seems that construct handles things recursively.

In the 2nd test, the structure.geometry would be raising validation errors because of the infinity if this was not the case.

Maybe you can see if this has to do with pyest version?

Otherwise, I feel like we might be ok using `construct` in the backend.

### CODE REVIEW

**Comment by momchil-flex** [comment]
These tests work because in the constructed sim, `len(sim.monitors)` and `len(sim.structures)` is 0. :)

Otherwise if this works at some point I think the two tests can be combined into one? The first one for example would already test if the monitors are loaded correctly as models.

File: tests/test_construct.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    # should not error if loading with construct
+    sim2 = td.Simulation.construct(sim_dict)
+
+    assert len(sim2.monitors) > 0, "monitor list not added"
+
+    # do something with the loaded sim
+    for monitor in sim2.monitors:
+        monitor.storage_size(10, np.arange(10))
+
+    assert sim == sim2
```
Comment (momchil-flex) [inline tests/test_construct.py:37]: `sim` is the one that raises an error if you try to init

Comment (tylerflex) [reply tests/test_construct.py]: thanks

**Comment by tylerflex** [comment]
we can reopen this PR when we get around to working on construct, I think it doesn't make sense to keep around so cloising

**Pull request closed without merging**